enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,184 @@
+log warnings/errors. Log files will be stored at c:\var\log\containerd
+
+## upgrade 
+When a cluster is upgraded OLM will switch to using a new Red Hat operators
+index. Because WMCO is named the same in both indexes, OLM will upgrade WMCO
+from the previous version, up to the latest version available in the new
+cluster.
+
+The procedure for an upgrade is as follows:
+1) As part of upgrade, basic validation will be done and de-configuring  will take place.
+2) During De-configuration, older version of kubelet, kube-proxy and CNI will be uninstalled.
+3) The newer version of WMCO will install all the required components along with containerd. 
+4) Kuebelet will start using containerd for pulling and managing container images. 
+
+   
+### Risks and Mitigations
+
+* If the cluster is upgraded, and the new version introduces an issue due to containerd
+  that should be addressed coz docker runtime support won't be available in kubelet.
+  going back to older WMCO might also run into issue due to kubelet version mismatch
+  between API server and kubelet.As we are planning to bring this feature before 4.11
+  this can be tested well and should make sure we address all the issues by working
+  with containerd open source community.
+* containerd doesn't support image-pull-progress-deadline as of now. There is a PR
+  https://github.com/containerd/containerd/pull/6150 work in progress. Until this 
+  gets merged, if windows image pull takes more time than the default value, we might
+  run into to image pull timeout error. proposed solution would be pull the image first
+  with ctr or crictl commandline tool and then create pods.
+* Currently windows_exporter has been used to collect metrics from windows node. we do 
+  have containerd support in https://github.com/prometheus-community/windows_exporter/releases/tag/v0.16.0
+  All the functionalities supported in docker runtime need to be checked for containerd.
+
+### Test Plan
+
+* We will have new e2e test case to be added for containerd replicating same existing 
+  test case that covers WMCO functionality. 
+* Containerd is agnostic to platform so testing in any platform should be fine.
+* Update WMCO community image on release repo so that CI workflow will use containerd
+  based WMCO community operator. 
+
+### Graduation Criteria
+
+This enhancement will start with WMCO community operator. This will become default feature from 
+openshift 4.11 release onwards.
+
+#### Dev Preview -> Tech Preview
+None
+
+#### Tech Preview -> GA
+None
+
+#### Removing a deprecated feature
+N/A
+
+### Upgrade / Downgrade Strategy
+
+Upgrade is already discussed in design section. Downgrades are [not supported](https://github.com/operator-framework/operator-lifecycle-manager/issues/1177)
+by OLM.
+
+### Version Skew Strategy
+We plan to maintain parity with the upstream [containerd](https://github.com/containerd/containerd/releases)
```

Comment:
Should we would mention how i.e submodule updates from [our downstream repo](https://github.com/openshift/containerd)